app/controllers/hyrax/dashboard/collections_controller.rb

```diff
@@ -583,6 +594,23 @@ def after_create_errors(errors)
           wants.json { render_json_response(response_type: :unprocessable_entity, options: { errors: errors }) }
         end
       end
+
+      def after_update_response
+        respond_to do |format|
+          format.html { redirect_to update_referer, notice: t('hyrax.dashboard.my.action.collection_update_success') }
+          format.json { render json: @collection, status: :updated, location: dashboard_collection_path(@collection) }
+        end
+      end
+
+      def after_update_errors(errors)
+        respond_to do |wants|
+          wants.html do
+            flash[:error] = errors.to_s
+            render 'edit', status: :unprocessable_entity
+          end
+          wants.json { render_json_response(response_type: :unprocessable_entity, options: { errors: errors }) }
```

Comment:
Returning json upon error isn't tested for collections or works.  Best I can determine, when `@collections.errors = "a collection error"`

`render json` produces...
```
"a collection error"
```

`render_json_response` produces...
```
      {
          "code": 422,
          "message": "Unprocessable Entity",
          "description": "The resource you attempted to modify cannot be modified according to your request.",
          "errors": "a collection error"
        }
```

Works use `render_json_response` for create and update.  I confirmed that they did before PR #5448 which made a similar refactor in works.  

Collections PR #5604 did change from using `render json` to `render_json_response` for `after_create_errors`.

Options for backward compatibility:
* OPTION 1: for all collection models, use `render json`.  This would require updating of `after_create_errors` back to using `render json`
* OPTION 2: use `render json` if `Hyrax.config.collection_class.is_a? ActiveFedora::Base` and use `render_json_response` if valkyrie collection
* OPTION 3: go ahead and return the more expressive `render_json_response` for all (not backward compatible since processing of errors for json would change)

I lean toward OPTION 2.  The more expressive response that is compatible with the response for works seems like the best long term approach.

Reply:
Thanks for the thorough write up!

This change sort of feels like a bugfix but the `render json` output is valid JSON so I think it is probably best to play it safe and go with option 2.  Do you plan on adding some tests for the JSON response body?

Reply:
I added tests as a way to explore what was going on.  So there will be tests for both create and update for the various json responses.